### PR TITLE
NDRS-735 - separate deploys and transfers in BlockHeader

### PR DESCRIPTION
**Pull request by dwerner**
This PR implements NDRS-735 - separate transfers and deploys in the block header.

### CODE REVIEW

File: node/src/types/block.rs [hunk 1]
```diff
@@ -560,6 +561,11 @@ impl BlockHeader {
         &self.deploy_hashes
     }
 
+    /// The list of deploy hashes included in the block.
```
Comment (Fraser999) [inline node/src/types/block.rs:564]:
```suggestion
    /// The list of transfer hashes included in the block.
```

File: node/src/types/block.rs [hunk 2]
```diff
@@ -1142,6 +1153,10 @@ pub(crate) mod json_compatibility {
         pub fn deploy_hashes(&self) -> &Vec<DeployHash> {
             &self.header.deploy_hashes
         }
+        /// Returns the hashes of the transfer `Deploy`s included in the `Block`.
```
Comment (Fraser999) [inline node/src/types/block.rs:1156]:
```suggestion

        /// Returns the hashes of the transfer `Deploy`s included in the `Block`.
```

**Review by Fraser999 — COMMENTED** [review]
Couple of nitpicks, otherwise LGTM.

**Review by Fraser999 — APPROVED** [review]

**Comment by goral09** [comment]
This breaks clarity right? Since the serialization is changed.

**Comment by dwerner** [comment]
> This breaks clarity right? Since the serialization is changed.

Good point @zie1ony - this may affect the event store.

**Pull request merged**